samples/bluetooth/iso_broadcast/src/main.c

```diff
@@ -197,14 +212,87 @@ int main(void)
 			net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
 			sys_put_le32(iso_send_count, iso_data);
 			net_buf_add_mem(buf, iso_data, sizeof(iso_data));
-			ret = bt_iso_chan_send(&bis_iso_chan[chan], buf, seq_num);
+
+			/* Ensure that first ISO data on all streams are given with sequence number
+			 * 0, so that the (Zephyr) Controller uses the same payload counter for
+			 * a given sequence number on all streams at an SDU interval.
+			 * Read Tx Sync interface returns error if no ISO data was given previously.
+			 */
+			ret = bt_iso_chan_get_tx_sync(&bis_iso_chan[chan], &tx_info);
+			if (ret) {
+				printk("Failed to get tx sync before send (%d)\n", ret);
+
+				/* Use sequence number 0, to synchronize current stream with the
+				 * current payload counter in the ISO group of streams.
+				 */
+				sn = 0U;
+
+			} else {
+				sn = seq_num;
+			}
+
+			/* Send ISO data with timestamp, use sequence number 0 for first ISO data on
+			 * any stream.
+			 */
+			if (sn) {
+				ret = bt_iso_chan_send_ts(&bis_iso_chan[chan], buf, sn, ts);
+			} else {
+				ret = bt_iso_chan_send(&bis_iso_chan[chan], buf, sn);
+			}
+
 			if (ret < 0) {
 				printk("Unable to broadcast data on channel %u"
 				       " : %d", chan, ret);
 				net_buf_unref(buf);
 				return 0;
 			}
 
+			/* Get Tx Sync, to verify that all streams have their ISO data at the same
+			 * timestamp.
+			 */
+			ret = bt_iso_chan_get_tx_sync(&bis_iso_chan[chan], &tx_info);
+			if (ret) {
+				printk("Failed to get tx sync after send (%d)\n", ret);
+				return 0;
+			}
+
+			/* Timestamp delta with the Controller */
+			if (!delta) {
+				delta = tx_info.ts - ts;
+			} else {
+				int32_t drift = ((delta - tx_info.ts + ts) / BIG_SDU_INTERVAL_US) /
+						1000000U;
+
+				/* NOTE: As the `ts` is just a increase in software in this sample,
+				 *       we will not see any drift.
+				 */
```

Comment:
Not sure what you mean by that. 

In some previous on-target tests, I've seen the tx_info.ts change by a value that was not the SDU interval, but up to 100 microseconds in difference. Of course that is "just" jitter and not "drift".

Reply:
> Not sure what you mean by that.

This code down in this c file: `ts += BIG_SDU_INTERVAL_US;` i.e. ts is not sourced from a clock/timer but just an increase done in software.